### ISSUE TITLE
Can we avoid "index read failed: EOF"?

### ISSUE THREAD

**Issue by campoy** [src-d/gitbase#661]
Whenever a dataset changes the indices on it becomes useless and trying to read them causes the server to respond with `index read failed: EOF`.

I'd like to avoid these issues and simply ignore the failing indexes rather than failing the whole request.

**Comment by ajnavarro** [comment]
Related: https://github.com/src-d/gitbase/issues/580

**Comment by ajnavarro** [comment]
In my opinion, ignore the index shouldn't be an option. You can delete it if your dataset changed and then create it again using `DELETE INDEX` statement.

WDYT @src-d/data-retrieval @smola @mcuadros ?

**Comment by smola** [comment]
@ajnavarro I agree. We need to define a supported way to update/ingest data in gitbase (https://github.com/src-d/gitbase/issues/580 https://github.com/src-d/feature-idea/issues/106), right now, I would consider this just unsupported and dropping index as the only workaround.

**Comment by kuba--** [comment]
As a temporary solution, we can keep checksum+timestamp whatever what let us figure out for what data set an index was created. Like @ajnavarro said, so far we're _static/read-only_ query frontend for git dataset. There is no auto-updates mechanism. It has to be done manually (delete + create).
On the other hand, all DBs which support non-read-only operations do it via SQL interface (insert/update/alter) and it always super unsafe to modify _raw dataset_.

**Comment by campoy** [comment]
I understand that this is not a final solution, but I'd rather see a warning somewhere telling me that the index is not valid anymore (so the request will be slower than expected) rather than having all queries fail.

This could even be done at startup time, so when the server starts checks for the integrity of the existing indexes and drops those that are not readable or up to date with the current data while maybe warning in red about it.

Not supporting changes in the disk is not really an option for us, unfortunately, but maybe a layer of abstraction could  be added so when a file changes on disk that change is transformed into a sequence of INSERT or UPDATE commands
